src/com/vivian/java/leetcode/TreeNode.java

```diff
@@ -0,0 +1,23 @@
+package com.vivian.java.leetcode;
+import com.vivian.java.collections.ArrayList;
+
+public class TreeNode<T> {
+    private T data;
+    private ArrayList<TreeNode<T>> child = new ArrayList<>();
```

Comment:
Same here, program to an interface, not an implementation. 
Change the reference type to List.

Reply:
done